src/CoamClient.js

```diff
@@ -405,8 +405,12 @@ class CoamClient {
         return this.__exec(data);
     }
 
-    getUserPermissionsForResourceType(principal, resourceType) {
-        let url = this.__buildUrl(`/auth/access-management/v1/principals/{{principal}}/permissions/{{resourceType}}`, {
+    getUserPermissionsForResourceType(principal, resourceType, include = null, permissionFilters = '') {
+        let urlPath = `/auth/access-management/v1/principals/{{principal}}/permissions/{{resourceType}}`;
+        if (include) {
```

Comment:
What happens if you specify `permissionFilter` but not `include`?

See lines [423](https://github.com/Cimpress/coam-client/blob/master/src/CoamClient.js#L423) which do something similar - there isn't a good helper function yet available for a list of optional query parameters.

Reply:
include is needed for this endpoint BCS True and False both has a functionality
False indicates not to include the given permission filter while True indicates to give only the listed permission filters

I can set default true but then I have to also check permission filter is not null BCS if you pass include in path parameter and send null permission filter then you will get zero results

Reply:
Yes, but `include === false` means to _exclude_ the permissions in the permission filter. So that has a functionality by itself.

From the documentation:

> Boolean that indicates whether to include or exclude permissions in the permissionFilter field. **A value of true will include the listed permissions in the response, false will exclude the specified permissions.** If not specified, will default to true.